src/spikeinterface/core/baserecording.py

```diff
@@ -921,11 +917,11 @@ def time_to_sample_index(self, time_s):
                 sample_index = time_s * self.sampling_frequency
             else:
                 sample_index = (time_s - self.t_start) * self.sampling_frequency
-            sample_index = round(sample_index)
+            sample_index = np.round(sample_index).astype(int)
```

Comment:
Flagging this change! Sorry if this is not relevant enough to this PR, but I thought that while I was working on time logic it would be good to fix this last small quality of life thing (vectorizing time_to_sample_index -- note that the scalar case still behaves the same).

Reply:
I guess this follows @h-mayorquin 's [comment](https://github.com/SpikeInterface/spikeinterface/pull/3119#pullrequestreview-2154668157)

Reply:
Interesting! This is nice, I think the only consideration is possible overflow for longer recordings, as `int64` is capped but python `int()` is not capped. @h-mayorquin has been focussing on this more, but looking at a quick example below it should be fine.

`int64` max value is `9,223,372,036,854,775,807`. If we take a neuropixels recording, continuous for 2 months (not unfeasible these days) we have `(30,000 * 60 * 60 * 24 * 60) = 165888000000`  (samples per s x seconds per minute x minutes per hour x hours per day x ~days in 2 month) (please check). But, maybe in 5 years people are sampling at 100 kHz and doing year long recordings 😆 we would have max index of `(3.1536e+12)`. So I think should be sufficient under all feasible uses, but something to consider.

Reply:
Interesting... wait, maybe I'm doing the math wrong, but don't we have:

```python
# int64 max val     | samples/sec | sec/min | min/hr | hr/day
9223372036854775807 /     100_000 /      60 /     60 /     24
# => 1_067_519_911.6730065 days
```
which is quite a long time? ([wolfram double check](https://www.wolframalpha.com/input?i=9223372036854775807+*+10+microseconds+%3D+%3F+days))